### DISCUSSION TITLE
useFieldArray recursive

### DISCUSSION THREAD

**Discussion by publicaccount365** [react-hook-form/react-hook-form#2555] [Q&A]
how to build form with nested inputs array?
for example:
```
[
    {
	paramName: "",
	paramDesc: "" ,
	parameters: [
		{ 
			 paramName:"",
			 paramDesc: "",
			 parameters: [
				{ 
				     paramName:"",
			    	     paramDesc: ""
				},
                                       ....
			]
		},
	]
    }
]
```

**Comment by bluebill1049** [comment]
https://github.com/react-hook-form/react-hook-form/tree/master/examples

**Array of Field Array**

**Reply by publicaccount365** [reply]
> https://github.com/react-hook-form/react-hook-form/tree/master/examples
> 
> **Array of Field Array**

thanks, but i want nested input array ( recursive  )
(example in above https://github.com/react-hook-form/react-hook-form/discussions/2555#discussion-13203). 
In example (Array of Field Array) use array with one level.

**Comment by publicaccount365** [comment]
No one has a solution?

**Reply by mdugue** [reply]
hey @publicaccount365  did you figure out a way to get it work? I found https://wanago.io/2022/05/16/recursive-dynamic-forms-react-hook-form-typescript/ but there is some smelly `as` Typescript overrides in there
